compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2867,8 +2878,8 @@ object Types {
     }
   }
 
-  case class LazyRef(private var refFn: Context => Type) extends UncachedProxyType with ValueType {
-    private var myRef: Type = null
+  case class LazyRef(private var refFn: (Context => (Type | Null)) | Null) extends UncachedProxyType with ValueType {
```

Comment:
Add a comment stating the invariant that `refFn` can be null only if `computed` is true (if that's the correct invariant).